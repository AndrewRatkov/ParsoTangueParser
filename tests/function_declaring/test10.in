def str abacaba(int x)
if abacaba(0) == "hrundelair"
then return "OK";
else
x := x + 1;
return "NOT OK"*x;
fi
enddef