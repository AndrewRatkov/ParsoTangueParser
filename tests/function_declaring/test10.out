[DEF FUN abacaba(INTEGER x)=>STR]
|   
+-->IF
    |
    +--->[INT,==]------->[CALL STR abacaba]------->[INT,0]
    |        |
    |        +---------->[STR,"hrundelair"]
    |
    THEN
    |
    +--->[RETURN]--->[STR,"OK"]
    |
    ELSE
    |
    +--->[DEF int x]--->[INT,+ ]------->[INT,x]
    |                       |
    |                       +---------->[INT,1]
    |
    +--->[RETURN]--->[STR,* ]------->[STR,"NOT OK"]
    |                    |
    |                    +---------->[INT,x]
    |
    FI
